Sources/StreamChat/Utils/MulticastDelegate.swift

```diff
@@ -36,30 +41,38 @@ struct MulticastDelegate<T> {
     /// Sets the main delegate. If is nil, removes the main delegate.
     /// - Parameter mainDelegate: The main delegate.
     mutating func set(mainDelegate: T?) {
-        _mainDelegate.removeAllObjects()
-
-        if let delegate = mainDelegate {
-            _mainDelegate.add(delegate as AnyObject)
+        queue.sync {
+            _mainDelegate.removeAllObjects()
+            
+            if let delegate = mainDelegate {
+                _mainDelegate.add(delegate as AnyObject)
+            }
```

Comment:
Why do we need to wait for this to complete? 🤔 Isn't it better to be async?
```suggestion
        queue.async {
            _mainDelegate.removeAllObjects()
            
            if let delegate = mainDelegate {
                _mainDelegate.add(delegate as AnyObject)
            }
```

Reply:
Since it is just a property mutation it does not really matter much. But we can keep it more aligned with other files where I think we typically async every mutation.

Reply:
It depends on how much times this is called. If it is called a lot of times, the `sync` could be problematic 🤔

Reply:
Whenever possible we should use `async` because `sync` increases the changes for deadlocks as well